projects/lights_control/src/lights_gpio.c

```diff
@@ -0,0 +1,212 @@
+      [LIGHTS_RIGHT_REAR_OUTER_TURN_PERIPHERAL] = { .port = GPIO_PORT_B, .pin = 0 },  //
+      [LIGHTS_RIGHT_REAR_TURN_PERIPHERAL] = { .port = GPIO_PORT_B, .pin = 10 },       //
+    };
+
+static const GPIOAddress s_board_type_address = { .port = GPIO_PORT_B, .pin = 13 };
+
+static LightsBoard s_board;
+
+static const GPIOAddress *s_address_front_lookup[] = {
+  [LIGHTS_EVENT_HORN] = s_horn_peripheral_addresses,                        //
+  [LIGHTS_EVENT_HIGH_BEAMS] = s_high_beams_peripheral_addresses,            //
+  [LIGHTS_EVENT_LOW_BEAMS] = s_low_beams_peripheral_addresses,              //
+  [LIGHTS_EVENT_DRL] = s_drl_peripheral_addresses,                          //
+  [LIGHTS_EVENT_SIGNAL_LEFT] = s_front_signal_left_peripheral_addresses,    //
+  [LIGHTS_EVENT_SIGNAL_RIGHT] = s_front_signal_right_peripheral_addresses,  //
+};
+
+static const GPIOAddress *s_address_rear_lookup[] = {
+  [LIGHTS_EVENT_BRAKES] = s_brakes_peripheral_addresses,
+  [LIGHTS_EVENT_STROBE] = s_strobe_peripheral_addresses,
+  [LIGHTS_EVENT_SIGNAL_LEFT] = s_rear_signal_left_peripheral_addresses,
+  [LIGHTS_EVENT_SIGNAL_RIGHT] = s_rear_signal_right_peripheral_addresses,
+};
+
+static StatusCode prv_init_peripheral(GPIOAddress *peripherals) {
+  uint8_t i;
+
+  // same settings are used for all ports
+  GPIOSettings settings = { .direction = GPIO_DIR_OUT,
+                            .state = GPIO_STATE_HIGH,
+                            .resistor = GPIO_RES_NONE,
+                            .alt_function = GPIO_ALTFN_NONE };
+
+  for (i = 0; i < SIZEOF_ARRAY(peripherals); i++) {
+    StatusCode status = gpio_init_pin(&peripherals[i], &settings);
+    if (status != STATUS_CODE_OK) {
+      return status;
+    }
+  }
+  return STATUS_CODE_OK;
+}
+
+StatusCode lights_gpio_init() {
+  uint8_t i;
+
+  const GPIOAddress **address_lookup;
+  address_lookup = (s_board == LIGHTS_BOARD_FRONT) ? s_address_front_lookup : s_address_rear_lookup;
+
+  for (i = 0; i < SIZEOF_ARRAY(address_lookup); i++) {
+    StatusCode status = prv_init_peripheral(address_lookup[i]);
+    if (status != STATUS_CODE_OK) {
+      return status;
+    }
+  }
+  return STATUS_CODE_OK;
+}
+
+StatusCode lights_gpio_get_lights_board(LightsBoard *board) {
+  GPIOSettings settings = { .direction = GPIO_DIR_OUT,
+                            .state = GPIO_STATE_HIGH,
+                            .resistor = GPIO_RES_NONE,
+                            .alt_function = GPIO_ALTFN_NONE };
+
+  // initializing the board type pin, before reading the board type from it
+  StatusCode init_status = gpio_init_pin(&s_board_type_address, &settings);
+  if (init_status != STATUS_CODE_OK) {
+    return init_status;
+  }
+
+  // reading the state to know the board type
+  GPIOState state;
+  StatusCode state_status = gpio_get_state(&s_board_type_address, &state);
+  if (state_status != STATUS_CODE_OK) {
+    return state_status;
+  }
+  if (state == GPIO_STATE_LOW) {
+    s_board = LIGHTS_BOARD_FRONT;
```

Comment:
Why do you have a static board? Why not just use the board instance?

Reply:
Some other modules need to know what board they're operating on. I've moved all this pin lookup to `lights_gpio_init`, and changed this function to only return the board type.